chalk-solve/src/lib.rs

```diff
@@ -61,16 +61,6 @@ pub trait RustIrDatabase {
     ) -> (Arc<AssociatedTyDatum>, &'p [Parameter], &'p [Parameter]);
 }
 
-/// The trait for defining the program clauses that are in scope when
-/// solving a goal.
-pub trait ChalkSolveDatabase: RustIrDatabase + Debug {
-    /// Returns a set of program clauses that could possibly match
-    /// `goal`. This can be any superset of the correct set, but the
-    /// more precise you can make it, the more efficient solving will
-    /// be.
-    fn program_clauses_that_could_match(&self, goal: &DomainGoal, vec: &mut Vec<ProgramClause>);
-}
-
 pub use solve::Solution;
```

Comment:
Would you mind exposing `solve::Guidance` here as well? It's rather hard to use `Solution` without being able to destructure `Guidance` ;) (I could of course create a PR, but since we're depending on this branch anyway, it's just the one line, and it would probably conflict with the deleted lines above...)

Reply:
Done :+1: